### PR TITLE
Use stack for dependency management

### PR DESCRIPTION
**Pull request by gfontenot**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,13 +15,12 @@ where the static site is built, and then the whole thing is deployed to
...
 
-You'll need to have `cabal` and `ghc` installed. If you're on OS X, you can do
+You'll need to have `stack` and `ghc` installed. If you're on OS X, you can do
 that with homebrew:
 
 ```
 % brew update
-% brew install ghc cabal-install
-% cabal install cabal-install
+% brew install ghc stack
```
Comment (gabebw) [inline README.md]: - The package is named `haskell-stack`, not `stack`
- `haskell-stack` depends on `ghc`, so you probably don't need to explicitly mention `ghc`

This line could thus be: `% brew install haskell-stack`

File: README.md [hunk 2]
```diff
@@ -15,13 +15,12 @@ where the static site is built, and then the whole thing is deployed to
 
 ### Dependencies ###
 
-You'll need to have `cabal` and `ghc` installed. If you're on OS X, you can do
+You'll need to have `stack` and `ghc` installed. If you're on OS X, you can do
```
Comment (pbrisbin) [inline README.md]: Are you not using a stack-installed GHC?

File: circle.yml [hunk 3]
```diff
@@ -1,16 +1,28 @@
...
+  cache_directories:
+    - "~/.stack"
 
 test:
   override:
-    - cabal run --  clean
-    - cabal run --  build
-    - cabal run --  check --internal-links
+    - stack build
+    - stack exec site -- clean
```
Comment (pbrisbin) [inline circle.yml]: Probably don't need this. Do you expect `_site` to have content in a fresh checkout?

**Comment by pbrisbin** [comment]
LGTM

File: README.md (hunk 2 above)
Comment (gfontenot) [inline README.md]: Should I just _always_ be using stack to install ghc?

File: circle.yml (hunk 3 above)
Comment (gfontenot) [inline circle.yml]: Nope! Good point.

File: README.md (hunk 2 above)
Comment (pbrisbin) [inline README.md]: [I do](https://github.com/pbrisbin/dotfiles/blob/master/tag-haskell/stack/config.yaml#L4).

**Pull request merged**